### PR TITLE
[16.0][IMP] base_wamas_ubl: Support negative float and raise error when missing telegram

### PR DESCRIPTION
**Pull request by QuocDuong1306**
The module has already supported to parse negative float in wamas to ubl [here](https://github.com/OCA/edi/blob/a61dbb6e951304538125ddabe3424766a187d692/base_wamas_ubl/lib/wamas/utils.py#L402)
```
>>> b = '-00000024000000000000000'
>>> dp =15
>>> c = float(b[:-dp] + "." + b[-dp:])
>>> c
-24.0
```

### CODE REVIEW

File: base_wamas_ubl/lib/wamas/utils.py [hunk 1]
```diff
@@ -91,17 +91,25 @@ def _set_string_int(val, length, dp, **kwargs):
 def _set_string_float(val, length, dp, **kwargs):
     res = str(float(val or 0))
```
Comment (nilshamerlinck) [inline base_wamas_ubl/lib/wamas/utils.py]: what about this line?

File: base_wamas_ubl/lib/wamas/tests/test_utils.py [hunk 2]
```diff
@@ -64,6 +65,15 @@ def test_set_string_bool(self):
         self.assertEqual(_set_string_bool("N", 1, False), "N")
         self.assertEqual(_set_string_bool("J", 1, False), "J")
 
+    def test_set_string_float(self):
+        # Input is the positive value
```
Comment (nilshamerlinck) [inline base_wamas_ubl/lib/wamas/tests/test_utils.py]:
```suggestion
        # positive
```

File: base_wamas_ubl/lib/wamas/tests/test_utils.py [hunk 3]
```diff
@@ -64,6 +65,15 @@ def test_set_string_bool(self):
...
+        self.assertEqual(_set_string_float(3.6, 9, 3), "000003600")
+        self.assertEqual(_set_string_float(3.0, 9, 3), "000003000")
+
+        # Input is the negative value
```
Comment (nilshamerlinck) [inline base_wamas_ubl/lib/wamas/tests/test_utils.py]:
```suggestion
        # negative
```

File: base_wamas_ubl/lib/wamas/utils.py [hunk 4]
```diff
@@ -91,17 +91,25 @@ def _set_string_int(val, length, dp, **kwargs):
...
+        res = str(float(val or 0))
+    except ValueError as err:
         raise Exception(
             "The value '%s' is not the float type. Please check it again!" % res
-        )
+        ) from err
+
+    # Support for the negative float
+    signed = ""
+    if "-" in res:
```
Comment (nilshamerlinck) [inline base_wamas_ubl/lib/wamas/utils.py]: res.startswith?

Comment (QuocDuong1306) [reply base_wamas_ubl/lib/wamas/utils.py]: I updated the PR

**Review by nilshamerlinck — APPROVED** [review]

**Pull request closed without merging**